### PR TITLE
Allows specification of custom config for Jest

### PR DESCRIPTION
**Pull request by evansiroky**
Following on the heels of https://github.com/conveyal/mastarm/pull/271, this adds the ability to load a file relative to the current working directory that contains JSON either as the default export of a _.js_ file or the contents of a _.json_ file.

### CODE REVIEW

**Comment by evansiroky** [comment]
Currently there are 2 ways to sort of configure Jest. There is either the [command line options](https://jestjs.io/docs/en/cli) and there are [configurations](https://jestjs.io/docs/en/configuration). Mastarm currently allows specifying custom cli arguments via #271. This PR now adds the ability to customize the configuration of jest. The CLI options and configuration options sometimes overlap and other times there are things specific to each.

**Comment by landonreed** [comment]
Please provide a way to test out the use of a custom config.

**Comment by evansiroky** [comment]
> Please provide a way to test out the use of a custom config.

I just refactored the documentation of mastarm test. Also, I had already augmented the existing test case in `__tests__/lib/jest.js` to show how this could be done.

File: __tests__/test-utils/mocks/mock-jest-config.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{ "coverage": true }
```
[the remark this answers is not in the document]
Comment (evansiroky) [reply __tests__/test-utils/mocks/mock-jest-config.json]: Yep, good catch.

**Comment by evansiroky** [comment]
Just did a small refactor to hopefully address your comments.

**Pull request merged**